web/src/features/fireWeather/components/tables/StationComparisonTable.tsx

```diff
@@ -0,0 +1,482 @@
+import {
+  TEMPERATURE_VALUES_DECIMAL,
+  RH_VALUES_DECIMAL,
+  WIND_SPEED_VALUES_DECIMAL,
+  PRECIP_VALUES_DECIMAL,
+  DEW_POINT_VALUES_DECIMAL
+} from 'utils/constants'
+
+const useStyles = makeStyles({
+  paper: {
+    padding: '5px'
+  },
+  typography: {},
+  lightColumnHeader: {
+    textAlign: 'center',
+    padding: '2px',
+    minWidth: '60px'
+  },
+  lightColumn: {
+    textAlign: 'right',
+    padding: '2px'
+  },
+  windSpeedValue: {
+    whiteSpace: 'nowrap'
+  },
+  relativeHumidityValue: {
+    whiteSpace: 'nowrap'
+  },
+  windDirectionValue: {
+    whiteSpace: 'nowrap'
+  },
+  precipitationValue: {
+    whiteSpace: 'nowrap'
+  },
+  darkColumn: {
+    backgroundColor: '#fafafa',
+    padding: '1px',
+    textAlign: 'right'
+  },
+  darkColumnHeader: {
+    backgroundColor: 'rgb(240, 240, 240)',
+    textAlign: 'center',
+    padding: '2px',
+    minWidth: '60px'
+  }
+})
+
+interface Props {
+  timeOfInterest: string
+  stationCodes: number[]
+  stationsByCode: Record<number, GeoJsonStation | undefined>
+  allNoonForecastsByStation: Record<number, NoonForecastValue[] | undefined>
+  observationsByStation: Record<number, ObservedValue[] | undefined>
+  allHighResModelsByStation: Record<number, ModelValue[] | undefined>
+  allRegionalModelsByStation: Record<number, ModelValue[] | undefined>
+  allModelsByStation: Record<number, ModelValue[] | undefined>
+}
+
+const findNoonMatch = (
+  noonDate: string,
+  collection: ModelValue[] | undefined
+): ModelValue | undefined => {
+  return collection?.find((item: ModelValue) => item.datetime === noonDate)
+}
+
+interface AccumulatedPrecipitation {
+  precipitation: number | undefined
+  modelValues: ModelValue[]
+}
+
+const calculateAccumulatedPrecip = (
+  noonDate: string,
+  collection: ModelValue[] | undefined
+): AccumulatedPrecipitation | undefined => {
+  // We are calculating the accumulated precipitation from 24 hours before noon.
+  const from = DateTime.fromISO(noonDate).toJSDate()
+  from.setHours(from.getHours() - 24)
+  const to = DateTime.fromISO(noonDate).toJSDate()
+  if (collection) {
+    const precip = {
+      precipitation: undefined,
+      modelValues: [] as ModelValue[]
+    } as AccumulatedPrecipitation
+    collection.forEach(value => {
+      const precipDate = DateTime.fromISO(value.datetime).toJSDate()
+      if (precipDate > from && precipDate <= to) {
+        if (typeof value.delta_precipitation === 'number') {
+          if (precip.precipitation === undefined) {
+            precip.precipitation = value.delta_precipitation
+          } else {
+            precip.precipitation += value.delta_precipitation
+          }
+          // Keep track of the model run predictions used to calculate this.
+          precip.modelValues.push(value)
+        }
+      }
+    })
+    return precip
+  }
+  return undefined
+}
+
+type TemperatureSourceType = NoonForecastValue | ObservedValue | ModelValue | undefined
```

Comment:
any special reason why you created a special type for `TemperatureSourceType` but not for the other weather variables?

Reply:
Sonarcloud or something complained that with that many types it was too long to stick straight into the function definition like the others.